pkg/controller/controller_serviceclass.go

```diff
@@ -33,30 +37,65 @@ func (c *controller) serviceClassAdd(obj interface{}) {
 	c.serviceClassQueue.Add(key)
 }
 
+func (c *controller) serviceClassUpdate(oldObj, newObj interface{}) {
+	c.serviceClassAdd(newObj)
+}
+
+func (c *controller) serviceClassDelete(obj interface{}) {
+	serviceClass, ok := obj.(*v1beta1.ClusterServiceClass)
+	if serviceClass == nil || !ok {
+		return
+	}
+
+	glog.V(4).Infof("Received delete event for ServiceClass %v; no further processing will occur", serviceClass.Name)
+}
+
 // reconcileServiceClassKey reconciles a ServiceClass due to controller resync
 // or an event on the ServiceClass.  Note that this is NOT the main
 // reconciliation loop for ServiceClass. ServiceClasses are primarily
 // reconciled in a separate flow when a ClusterServiceBroker is reconciled.
 func (c *controller) reconcileClusterServiceClassKey(key string) error {
-	// currently, this is a no-op.  In the future, we'll maintain status
-	// information here.
-	return nil
+	plan, err := c.serviceClassLister.Get(key)
+	if errors.IsNotFound(err) {
```

Comment:
I still think that we should check for error type only if `err != nil`, it's cleaner.
Thus, I would rewrite it to
```go
if err != nil {
  if errors.IsNotFound(err) {
    ...
    return nil
  }
  ...
  return err
}
```
Not sure why `errors.IsNotFound()` silently accepts `nil`.